tasks/pdp/proofset_addroot_watch.go

```diff
@@ -59,7 +59,7 @@ func processPendingProofSetRootAdds(ctx context.Context, db *harmonydb.DB, ethCl
 	err := db.Select(ctx, &rootAdds, `
         SELECT DISTINCT proofset, add_message_hash
         FROM pdp_proofset_root_adds
-        WHERE add_message_ok = TRUE
+        WHERE add_message_ok = TRUE AND roots_added = FALSE
```

Comment:
@rvagg Same as below, why do you need to filter on `roots_added = FALSE` ? This update is idempotent.

Reply:
Cause I don't want to do extra work. Strictly yes, it'd be idempotent, but we know the limiting condition here so we can `if len(rootAdds) == 0 {` and bail early if there's nothing and we don't spin our wheels.

Reply:
Remember the code before this would return `len(rootAdds)` because they were deleted previously